### ISSUE TITLE
Issues with rendering an `@interface`

### ISSUE THREAD

**Issue by ehallander9591** [jsdoc3/jsdoc#1406]
Using protocol buffers for model management, and the latest release changed from a typedef specification to an interface specification. The jsdoc generated documentation had output files for both typedef, and the classdesc described sections of each protocol buffer type. The typedefs would show up as belonging to the namespace that they were a member of, in this case mpxs.model

However, when it changed to @interface, the interface spec doesn't render as a file.

### Input code

When the output used typedef each message would look similar to:

            /**
             * Properties of a DecisionCriteria.
             * @typedef mpxs.model.DecisionCriteria$Properties
             * @type {Object}
             * @property {string} [id] (RO) ID of the criteria
             * @property {string} [missionId] (RO) ID for which the criteria apply
             * @property {Array.<mpxs.model.DecisionCriteriaItem$Properties>} [decisionCriteriaItem] (W) A single criterion
             * @property {Array.<mpxs.model.Metadata$Properties>} [metadata] (W) Metadata for the analysis overall
             */

            /**
             * Constructs a new DecisionCriteria.
             * @classdesc Criteria for a decision
             * @exports mpxs.model.DecisionCriteria
             * @constructor
             * @param {mpxs.model.DecisionCriteria$Properties=} [properties] Properties to set
             */
            function DecisionCriteria(properties) {
                this.decisionCriteriaItem = [];
                this.metadata = [];
                if (properties)
                    for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                        if (properties[keys[i]] != null)
                            this[keys[i]] = properties[keys[i]];
            }

and after the change to @interface the above changed to

            /**
             * Properties of a DecisionCriteria.
             * @memberof mpxs.model
             * @interface IDecisionCriteria
             * @property {string} [id] (RO) ID of the criteria
             * @property {string} [missionId] (RO) ID for which the criteria apply
             * @property {Array.<mpxs.model.IDecisionCriteriaItem>} [decisionCriteriaItem] (W) A single criterion
             * @property {Array.<mpxs.model.IMetadata>} [metadata] (W) Metadata for the analysis overall
             */

            /**
             * Constructs a new DecisionCriteria.
             * @memberof mpxs.model
             * @classdesc Criteria for a decision
             * @constructor
             * @param {mpxs.model.IDecisionCriteria=} [properties] Properties to set
             */
            function DecisionCriteria(properties) {
                this.decisionCriteriaItem = [];
                this.metadata = [];
                if (properties)
                    for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                        if (properties[keys[i]] != null)
                            this[keys[i]] = properties[keys[i]];
            }

Just for completeness sakes, this definition is preceded by 

    /**
     * Namespace mpxs.
     * @exports mpxs
     * @namespace
     */
    var mpxs = {};

    mpxs.model = (function() {

        /**
         * Namespace model.
         * @memberof mpxs
         * @namespace
         */
        var model = {};

to define the enclosing namespace.

```js
// No Code to provide
```

### JSDoc configuration

```js
{
    "tags": {
        "allowUnknownTags": false
    },
    "source": {
        "include": [
            "./public/lib/mpxsStaticModule.js"
        ],
        "exclude": [],
        "includePattern": ".+\\.js(doc)?$",
        "excludePattern": "(^|\\/|\\\\)_"
    },
    "plugins": [
        "./node_modules/jsdoc/plugins/markdown"
    ],
    "templates": {
        "cleverLinks"   : false,
        "monospaceLinks": false,
        "default"       : {
            "outputSourceFiles" : false
        },
        "applicationName": "mpxs.js",
        "disqus": "",
        "googleAnalytics": "",
        "openGraph": {
            "title": "",
            "type": "website",
            "image": "",
            "site_name": "",
            "url": ""
        },
        "meta": {
            "title": "",
            "description": "",
            "keyword": ""
        },
        "linenums"      : true
    },
    "markdown"  : {
        "parser"        : "gfm",
        "hardwrap"      : true
    },
    "opts": {
        "encoding"      : "utf8",
        "recurse"       : true,
        "private"       : false,
        "lenient"       : false,
        "destination"   : "./public/protobuf",
        "template"      : "./node_modules/jaguarjs-jsdoc",
        "sourceRoot"    : "MPXSCore",
        "repo"          : "MPXSCore.git"
    }
}
```

### JSDoc debug output

I don't get any output of note from running jsdoc.

```
Your debug output here
```

### Your environment

| Software         | Version
| ---------------- | -------
| JSDoc            | 3.5.2
| Node.js          | 6.10.2
| npm              | 3.10.10
| Operating system | Windows 10

**Comment by hegemonic** [comment]
This is an issue with the `jaguarjs-jsdoc` template. I just took a look at the template, and it doesn't appear to support interfaces.

JSDoc's default template generates an output page for the interface, as expected. However, the page for the `mpxs.model` namespace doesn't list the interfaces that belong to that namespace, which is a bug. I'm going to keep this issue open to deal with that bug.

**Comment by hegemonic** [comment]
The issue with the default template is fixed on master and releases/3.5. The fix will be included in JSDoc 3.5.3.

**Comment by ehallander9591** [comment]
Many thanks. I will take this over to jaguar-jsdoc
